ixmp/core.py

```diff
@@ -1000,8 +1017,9 @@ def clear_cache(self, name=None):
         if name is None:
             self._pycache = {}
         # remove this element from the python data cache
-        if name is not None and name in self._pycache:
-            del self._pycache[name]
+        for key in self._pycache:
```

Comment:
This will result in an error. An improvement would be:
self._pycache.pop(key for key in self._pycache.keys() if (self._pycache[key] == name)][0])